### PR TITLE
Fix Home rows after watched and dismissal actions

### PR DESCRIPTION
**Pull request by Quick104**
## Summary
- move Continue Watching dismissal and watched-state mutations into `HomeViewModel`
- update visible Home sections and the shared cache after successful mutations
- remove completed items immediately from Continue Watching, In Progress, and Next Up, then reconcile replacement episodes from the server
- invalidate stale in-flight Home requests and roll back card state when a watched update fails
- share the behavior across iOS and tvOS Home feeds

## Testing
- `xcodebuild test -project Silo.xcodeproj -scheme Silo -destination 'platform=iOS Simulator,id=5AEC2DBA-2CA4-4AA0-8EC8-333376BA51BC' -only-testing:SiloTests/HomeSectionsMutationTests CODE_SIGNING_ALLOWED=NO` (9 tests passed)
- `xcodebuild build -project Silo.xcodeproj -scheme Silo -destination 'platform=iOS Simulator,id=5AEC2DBA-2CA4-4AA0-8EC8-333376BA51BC' CODE_SIGNING_ALLOWED=NO`
- `xcodebuild build -project Silo.xcodeproj -scheme SiloTV -destination 'platform=tvOS Simulator,id=462E76D9-A6F7-4CBA-B8A8-1274FB2CB00B' CODE_SIGNING_ALLOWED=NO`
- installed and launched the resulting iOS build on `Silo Validation iPhone`

## Summary by CodeRabbit

* **New Features**
  * Enabled Continue Watching item dismissal and watched/unwatched updates across iOS and tvOS Home.
  * Actions now optimistically update the UI and keep Home in sync after server reconciliation.
  * Added user-facing alert when an item action can’t be completed.
* **Bug Fixes**
  * Reverted UI state when watched/dismiss actions fail.
  * Prevented older Home section responses from overwriting newer updates.
* **Tests**
  * Added unit/async coverage for removal, watched-state changes, duplicates, and failure/edge cases.

### CODE REVIEW

File: iosApp/iosApp/Screens/Home/SectionRow.swift [hunk 1]
```diff
@@ -8,6 +8,8 @@ struct SectionRow: View {
     let section: ResolvedSection
     let onItemTap: (String) -> Void
     var onSeeAll: (() -> Void)? = nil
+    var onRemoveFromContinueWatching: ((SectionItem) -> Void)? = nil
+    var onSetWatched: ((SectionItem, Bool) async -> Bool)? = nil
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Screens/Home/SectionRow.swift:12]: Fixed in 2bb4223. SectionRow now always supplies a watched action: Home uses the injected model-owned mutation, while Recommendations, Browse, and tvOS library rows fall back to the original ContinuumAPI.setWatched behavior. The async result also restores the card state on failure. Revalidated with 9 focused iOS tests and a successful tvOS build.
